api/v1beta1/hostedcluster_types.go

```diff
@@ -733,6 +741,17 @@ type PowerVSPlatformSpec struct {
 	StorageOperatorCloudCreds corev1.LocalObjectReference `json:"storageOperatorCloudCreds"`
 }
 
+type KubeVirtPlatformSpec struct {
+	// ExtClusterName is the name of the external infra cluster used to host the guest cluster nodes
+	//
+	// +immutable
+	ExtClusterName string `json:"externalClusterName,omitempty"`
+	// KubeConfigSecret is the secret name that contains the kubeconfig for the external infra cluster
+	//
+	// +immutable
+	KubeConfigSecret *KubeconfigSecretRef `json:"kubeConfigSecret,omitempty"`
```

Comment:
I'm unsure if we need `ExtClusterName`. The secret ref should have the name of the secret that holds the creds, which should be all that's necessary.

In the `example.go` file, the unique secret can be generated using the hosted cluster name and a prefix, similar to how azure generates its credential secret

So, something like this in example.go for the kubevirt secret.
```
Name:      o.Name + "-infra-credentials",
```

Then when copying that over to the hosted control plane namespace, the credential name should be in the hostedcluster.platform.kubevirt.kubeConfigSecret ref just as you have it now.

Reply:
fixed.